### PR TITLE
Push ai-prompt-flow extension

### PR DESCRIPTION
**Pull request by Rubinuspy**
Difference from main repo
* Removed mermaid
* Removed node_modules and build.bazel
* Removed any symlink to the main repo

### CODE REVIEW

File: ai-prompt-flow/editorextensions/ai-prompt-flow/angular/projects/leftpanel/src/app/app.component.spec.ts
Comment (jjoneslucid) [inline ai-prompt-flow/editorextensions/ai-prompt-flow/angular/projects/leftpanel/src/app/app.component.spec.ts:1]: We should remove any tests that don't currently pass. They weren't being run with builds, so some of them are very out of date.

File: ai-prompt-flow/editorextensions/ai-prompt-flow/src/executor/promptexecutor.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    ) {
+        if (upstreamExecutions.has(system.id)) {
+            throw new Error(`Infinite loop detected with System Block: ${system.id}`);
+        } else if (upstreamExecutions.size > 10) {
+            // This measures depth, not breadth. This won't interfere with creating multiple
+            // branches, only with many models in a single chain. I can't think of a good reason to
+            // have more than 10 models in a single chain, but it is a pretty arbitrary limit so I
+            // don't have issues with bumping it up if there are valid uses.
+            throw new Error(
+                'Too many system blocks executed in a single session. Reach out to ML/AI 3 if you have a valid use case for this.',
```
Comment (jjoneslucid) [inline ai-prompt-flow/editorextensions/ai-prompt-flow/src/executor/promptexecutor.ts]: Not meaningful to anyone outside of Lucid, but also not harmful.

File: ai-prompt-flow/editorextensions/ai-prompt-flow/src/leftpanel.ts [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+                'canEditPackageSettings': canEditPackageSettings,
+            });
+        } else if (message.message == 'feedback') {
+            const beaconEvent = {
+                'name': 'sentAiFeedback',
+                'aiInteractionType': 'used ai prompt flow',
+                'rating': message.rating >= 4 ? 'thumbs up' : 'thumbs down',
+                'comment': 'Score: ' + message.rating + '/5. Comment: ' + message.feedback,
+            };
+            this.client.sendCommand(CommandName.FireBeaconEvent, {'e': JSON.stringify(beaconEvent)});
```
Comment (jjoneslucid) [inline ai-prompt-flow/editorextensions/ai-prompt-flow/src/leftpanel.ts]: We should replace this with `console.log(...)` or similar. The `FireBeaconEvent` command is only going to work for first-party extensions.

File: ai-prompt-flow/editorextensions/ai-prompt-flow/src/rightpanel/rightpanel.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    public async sendSelectionToPanel() {
+        const items = this.viewport
+            .getSelectedItems()
+            .map((item) => {
+                return convertToAiShape(item, this.client);
+            })
+            .filter(isDef);
+        if (items.length > 0) {
+            // Keep the right panel open whenever AI shapes are selected.
```
Comment (jjoneslucid) [inline ai-prompt-flow/editorextensions/ai-prompt-flow/src/rightpanel/rightpanel.ts]: nit: No longer applies, the right panel now gives priority to the extension panel. This doesn't do any harm, but isn't necessary.

File: ai-prompt-flow/editorextensions/ai-prompt-flow/src/blocks/blocks.ts [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+        this.deleteRow(editableRow + this.numHeaderRows);
+    }
+}
+
+function addRow(table: TableBlockProxy, index: number = table.getRows().length - 1) {
+    const {keyCell: cell} = getKeyValuePairFromTable(index, table);
+    const currentStyles = cell.getTextStyle();
+    const newRow = table.addRow(cell, false);
+    for (const newCell of newRow.getCells()) {
+        // See https://lucidsoftware.slack.com/archives/C01BTPAMZNJ/p1695334462696559
```
Comment (jjoneslucid) [inline ai-prompt-flow/editorextensions/ai-prompt-flow/src/blocks/blocks.ts]: The slack link won't be helpful to anyone outside of Lucid. The tl;dr should be enough by itself.

File: ai-prompt-flow/editorextensions/ai-prompt-flow/angular/projects/leftpanel/src/app/app.component.spec.ts
Comment (jjoneslucid) [reply ai-prompt-flow/editorextensions/ai-prompt-flow/angular/projects/leftpanel/src/app/app.component.spec.ts:1]: Actually looks like this is the only one that fails. The rightpanel tests are all passing.

**Comment by jjoneslucid** [comment]
Confirmed that the extension runs locally when using only the files in this PR.

**Review by jjoneslucid — APPROVED** [review]

**Review by blackburn32 — APPROVED** [review]

**Pull request merged**